src/crimson/os/seastore/transaction_manager.cc

```diff
@@ -247,18 +247,20 @@ TransactionManager::submit_transaction_direct(
       DEBUGT("journal commit to {} seq {}", tref, addr, journal_seq);
       segment_cleaner->set_journal_head(journal_seq);
       cache->complete_commit(tref, addr, journal_seq, segment_cleaner.get());
-      lba_manager->complete_transaction(tref);
-      segment_cleaner->update_journal_tail_target(
-	cache->get_oldest_dirty_from().value_or(journal_seq));
-      auto to_release = tref.get_segment_to_release();
-      if (to_release != NULL_SEG_ID) {
-	return segment_manager.release(to_release
-	).safe_then([this, to_release] {
-	  segment_cleaner->mark_segment_released(to_release);
-	});
-      } else {
-	return SegmentManager::release_ertr::now();
-      }
+      return lba_manager->complete_transaction(tref).safe_then(
```

Comment:
Maybe we can use when_all_succeeded to parallelize complete_transaction and other code that follows

Reply:
could you be more specific? why can we parallelize `complete_transaction()` and `segment_manager.release()` ? but i don't think the proposed change is in the scope of this changeset.

Reply:
I mean something like:

```
auto fut = lba_manager->complete_transaction(tref);
auto fut2 = SegmentManager::release_ertr::now();
segment_cleaner->update_journal_tail_target(
    cache->get_oldest_dirty_from().value_or(journal_seq)); 
auto to_release = tref.get_segment_to_release(); 
if (to_release != NULL_SEG_ID) {
    fut2 = segment_manager.release(to_release).safe_then([this, to_release] {
         segment_cleaner-&gt;mark_segment_released(to_release);
    }); 
}

return seastar::when_all_succeeded(fut, fu2);
```

is this feasible? :-)

Reply:
as discussed offline, i'd leave it to another PR after checking with Sam on the semantic of `complete_transaction()` and `segment_manager.release()`.